x-pack/plugins/apm/public/services/rest/savedObjects.ts

```diff
@@ -16,25 +15,13 @@ export interface ISavedObject {
   type: string;
 }
 
-interface ISavedObjectAPIResponse {
-  saved_objects: ISavedObject[];
-}
-
 export const getAPMIndexPattern = memoize(async () => {
-  const apmIndexPatternTitle: string = chrome.getInjected(
-    'apmIndexPatternTitle'
-  );
-  const res = await callApi<ISavedObjectAPIResponse>({
-    pathname: `/api/saved_objects/_find`,
-    query: {
-      type: 'index-pattern',
-      search: `"${apmIndexPatternTitle}"`,
-      search_fields: 'title',
-      per_page: 200
-    }
-  });
-
-  return res.saved_objects.find(
-    savedObject => savedObject.attributes.title === apmIndexPatternTitle
-  );
+  try {
+    return await callApi<ISavedObject>({
+      method: 'GET',
+      pathname: `/api/apm/index_pattern`
+    });
+  } catch (error) {
+    return;
```

Comment:
Do we want to swallow errors here? It seems like it might be a better UX to show an EuiCallOut with an error message (e.g. "APM failed to initialize" along with the status code and error message), or maybe just a toast notification. This way a user can report this info to support or the Discuss forums.

Reply:
This doesn't happen as part of initialization, but rather attempts to lazily create the APM index pattern when the UI attempts to load it. This memoized function is expected to return undefined if no index pattern is present (in this case, there was a failure to lazily create it). When undefined, the UI shows messaging to the user that the APM index pattern is required to use certain features, and the user is directed to explicitly create it.